pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+                common.append(prop_name)
+                prop = etree.SubElement(common_properties, 'property')
+                prop.set('description', '')
+                prop.set('name', prop_name)
+                prop.set('units', str(quantity.units))
+
+                value = etree.SubElement(prop, 'value')
+                value.text = str(quantity.magnitude)
+
+    # Ignition delay can't be common, unless only a single datapoint.
+
+    datagroup = etree.SubElement(root, 'dataGroup')
+    datagroup.set('id', 'dg1')
+    datagroup_link = etree.SubElement(datagroup, 'dataGroupLink')
+    datagroup_link.set('dataGroupID', '')
+    datagroup_link.set('dataPointID', '')
+
+    property_idx = {}
+    labels = {'temperature': 'T', 'pressure': 'P',
+              'ignition delay': 'tau', 'pressure rise': 'dP/dt',
+              }
+
+    for prop_name in ['temperature', 'pressure', 'ignition delay', 'pressure rise']:
+        attribute = prop_name.replace(' ', '_')
+        if (prop_name not in common and
+            any([getattr(dp, attribute, None) for dp in c.datapoints])
+            ):
+            prop = etree.SubElement(datagroup, 'property')
+            prop.set('description', '')
+            prop.set('name', prop_name)
+            prop.set('units', str(getattr(c.datapoints[0], attribute).units))
+            idx = 'x{}'.format(len(property_idx) + 1)
+            property_idx[idx] = prop_name
+            prop.set('id', idx)
+            prop.set('label', labels[prop_name])
+
+    if 'composition' not in common:
+        for species in c.datapoints[0].composition:
+            prop = etree.SubElement(datagroup, 'property')
+            prop.set('description', '')
+
+            idx = 'x{}'.format(len(property_idx) + 1)
+            property_idx[idx] = species['species-name']
+            prop.set('id', idx)
+            prop.set('label', '[' + species['species-name'] + ']')
+            prop.set('name', 'composition')
+            prop.set('units', c.datapoints[0].composition_type)
+
+            species_link = etree.SubElement(prop, 'speciesLink')
+            species_link.set('preferredKey', species['species-name'])
+            if species.get('InChI'):
+                species_link.set('InChI', species['InChI'])
+
+    for dp in c.datapoints:
+        datapoint = etree.SubElement(datagroup, 'dataPoint')
+        for idx in property_idx:
+            value = etree.SubElement(datapoint, idx)
+            value.text = str(getattr(dp, property_idx[idx].replace(' ', '_')).magnitude)
+
+    # if RCM and has volume history, need a second dataGroup
+    if (len(c.datapoints) > 1 and
+        any([getattr(dp, 'volume_history', None) for dp in c.datapoints])
+        ):
+        raise NotImplementedError('Error: ReSpecTh files do not support multiple datapoints '
+                                  'with a volume history.'
+                                  )
+        # TODO: what if they share the same history? Does this happen?
```

Comment:
Not in my experience, unless someone is taking a shortcut (which we should not support, IMHO)